### ISSUE TITLE
Dropout applied when computing test WER

### ISSUE THREAD

**Issue by samgd** [mozilla/DeepSpeech#1262]
During a run started by a command such as:

```
#!/bin/sh

python -u DeepSpeech.py \
  --train_files ... \
  --dev_files ... \
  --test_files ... \
  --train_batch_size 12 \
  --dev_batch_size 8 \ 
  --test_batch_size 8 \ 
  --epoch 13 \
  --learning_rate 0.0001 \
  --display_step 0 \ 
  --validation_step 1 \ 
  --dropout_rate 0.2367 \
  --default_stddev 0.046875 \
  --checkpoint_step 1 \ 
  --log_level 0 \ 
  --checkpoint_dir "${COMPUTE_KEEP_DIR}" \
  --wer_log_pattern "GLOBAL LOG: logwer('${COMPUTE_ID}', '%s', '%s', %f)"\
  --decoder_library_path "/usr/local/tensorflow1.4/lib/libctc_decoder_with_kenlm.so" \
  "$@"
```

The code will cycle through training and validation jobs before computing a WER based on the test data. However, it reuses the graph which has dropout applied still thus the WER is higher than expected.

This can be verified by comparing the WER at the end of a training run vs that of a test run only with `--dropout 0.0`.

There is a line of code that checks if `optimizer is None` and sets `no_dropout` if so but this will not be triggered during a training run:

https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L640

**Comment by reuben** [comment]
In the mean time, this is fixed by #1392.
